.github/workflows/codeql.yml

```diff
@@ -0,0 +1,81 @@
+name: "CodeQL"
+
+on:
+  push:
+    paths:
+      - '**/*.c'
+      - '**/*.cpp'
+      - '**/*.h'
+      - '**/*.hpp'
+      - '**/*.py'
+  pull_request:
+    paths:
+      - '**/*.c'
+      - '**/*.cpp'
+      - '**/*.h'
+      - '**/*.hpp'
+      - '**/*.py'
+
+jobs:
+  analyze-ubuntu:
+    name: Analyze on Ubuntu
+    runs-on: ubuntu-latest
+    permissions:
+      security-events: write
+      actions: read
+      contents: read
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp', 'python' ]
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v2
+      with:
+        languages: ${{ matrix.language }}
+
+    - name: Autobuild
+      uses: github/codeql-action/autobuild@v2
```

Comment:
why autobuild and not a custom one (like in other jobs)? you may have no control of used CMake params, I guess...?

Reply:
Right, we should build with all optional things enabled (like we do for other CI jobs).

Reply:
I analyzed params used in cmake.yml and I think that there is no param worth explicit set for CodeQL scan (for linux build at least), hence the Autobuild:
```
CMAKE_C_COMPILER/CMAKE_CXX_COMPILER - I think that building with any compiler should suffice,
UR_ENABLE_TRACING=ON - triggers third-party lib build, no need to scan that,
UR_DEVELOPER_MODE=ON - I assume that triggering additional compiler warnings do not affect CodeQL scan,
CMAKE_BUILD_TYPE - one of Debug/Release should suffice, chose Release, again only additional compiler flags are set in Debug mode,
UR_BUILD_TESTS=ON - worth having enabled, but it's enabled by default anyway,
UR_FORMAT_CPP_STYLE=ON - linting shouldn't affect this workflow
```

Reply:
also, I can see there's plan `make` command executed - could we make use of parallel building, like `make -j$(nproc)` ?